### ISSUE TITLE
T Crashes "End of stream receive cycle because of err="EOF", waiting for running transcode jobs to complete"

### ISSUE THREAD

**Issue by Titan-Node** [livepeer/go-livepeer#2702]
**Describe the bug**
A regular T crashes after a few days of being online.
This didn't happen before, not sure whats causing it. Been transcoding for over a year with this machine.

```
I1219 13:00:48.901495    9080 ot_rpc.go:128] End of stream receive cycle because of err="EOF", waiting for running transcode jobs to complete
```

Version 0.5.35

**Comment by leszko** [comment]
Thanks for reporting @Titan-Node 

Do you maybe have any steps to reproduce? Or it just happens from time to time without any specific reason?

**Comment by Titan-Node** [comment]
@leszko trying to reproduce but it happens so infrequently. 
Also I should mention it doesn't crash but instead ends up in a state where the program stays alive but does not do any more transcoding and disconnects from the O.

**Comment by eliteprox** [comment]
Yesterday I observed a recurrence of related issue https://github.com/livepeer/go-livepeer/issues/2605 which I thought was solved in the 0.5.37 release. Three transcoders connected to an O, one is intentionally disconnected and the O loses all streams, `no transcoders available`. Transcoders map went empty. I will try to reproduce the issue in a test B/O/T configuration as soon as possible.

**Comment by Titan-Node** [comment]
I believe this issue was fixed in this PR. https://github.com/livepeer/go-livepeer/pull/2747
Haven't seen this issue again since the update.
Thanks @eliteprox
